ogr/ogrsf_frmts/miramon/ogrmiramonlayer.cpp

```diff
@@ -0,0 +1,2613 @@
+                        }
+                        // It's a serialized JSON array
+                        else if (phMiraMonLayer->iMultiRecord == -3)
+                        {
+                            oField.SetType(OFTString);
+                            oField.SetSubType(OFSTJSON);
+                        }
+                        else  // iMultiRecord decides which Record translate
+                            oField.SetType(OFTString);
+                    }
+                    else if (phMiraMonLayer->pMMBDXP->pField[nIField]
+                                 .FieldType == 'N')
+                    {
+                        // It's a list?
+                        if (phMiraMonLayer->iMultiRecord == -2)
+                        {
+                            if (phMiraMonLayer->pMMBDXP->pField[nIField]
+                                    .DecimalsIfFloat)
+                                oField.SetType(phMiraMonLayer->isListField
+                                                   ? OFTRealList
+                                                   : OFTReal);
+                            else
+                                oField.SetType(phMiraMonLayer->isListField
+                                                   ? OFTIntegerList
+                                                   : OFTInteger);
+                        }
+                        // It's a serialized JSON array
+                        else if (phMiraMonLayer->iMultiRecord == -3)
+                        {
+                            oField.SetType(OFTString);
+                            oField.SetSubType(OFSTJSON);
+                        }
+                        else
+                        {
+                            if (phMiraMonLayer->pMMBDXP->pField[nIField]
+                                    .DecimalsIfFloat)
+                                oField.SetType(OFTReal);
+                            else
+                                oField.SetType(OFTInteger);
+                        }
+                    }
+                    else if (phMiraMonLayer->pMMBDXP->pField[nIField]
+                                 .FieldType == 'D')
+                    {
+                        // It's a serialized JSON array
+                        oField.SetType(OFTDate);
+                        if (phMiraMonLayer->iMultiRecord == -3)
+                        {
+                            oField.SetType(OFTString);
+                            oField.SetSubType(OFSTJSON);
+                        }
+                    }
+
+                    oField.SetWidth(
+                        phMiraMonLayer->pMMBDXP->pField[nIField].BytesPerField);
+                    oField.SetPrecision(phMiraMonLayer->pMMBDXP->pField[nIField]
+                                            .DecimalsIfFloat);
+
+                    poFeatureDefn->AddFieldDefn(&oField);
+                }
+            }
+        }
+
+        if (poSRS)
```

Comment:
can this poSRS be not null in this read code path ? I don't think so, otherwise it would override m_poSRS set at line 333 (and leak memory)

Reply:
poSRS has not to be used here.
Instead of 
        if (poSRS)
        {
            m_poSRS = poSRS->Clone();
            m_poSRS->SetAxisMappingStrategy(OAMS_TRADITIONAL_GIS_ORDER);
        }

        poFeatureDefn->GetGeomFieldDefn(0)->SetSpatialRef(m_poSRS);

It would be better
        m_poSRS->SetAxisMappingStrategy(OAMS_TRADITIONAL_GIS_ORDER);
        poFeatureDefn->GetGeomFieldDefn(0)->SetSpatialRef(m_poSRS);
        
  I leave this last code. I think it was a code from other part copyed here.